packages/terra-data-grid/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -153,14 +189,46 @@ const ColumnHeaderCell = (props) => {
     switch (key) {
       case KeyCode.KEY_SPACE:
       case KeyCode.KEY_RETURN:
-        onColumnSelect(id, { row: rowIndex, col: columnIndex });
+        if (isSelectable && onColumnSelect) {
+          onColumnSelect(id, { row: rowIndex, col: columnIndex }, isSelectable);
+        }
         event.stopPropagation();
         event.preventDefault(); // prevent the default scrolling
         break;
+      case KeyCode.KEY_LEFT:
+        if (isResizable && isResizeHandleActive) {
+          columnHeaderCellRef.current.focus();
+          setResizeHandleActive(false);
+
+          event.stopPropagation();
+          event.preventDefault();
+        }
+        break;
+      case KeyCode.KEY_RIGHT:
+        if (isResizable && !isResizeHandleActive) {
+          setResizeHandleActive(true);
+          event.stopPropagation();
+          event.preventDefault();
+        }
+        break;
       default:
     }
   };
 
+  const getTableHeight = () => {
+    if (columnHeaderCellRef.current) {
+      // Find parent table element
+      const parentTable = columnHeaderCellRef.current.closest('table');
+
+      // Update resize handle height to match parent table height
+      if (parentTable) {
+        return `${parentTable.offsetHeight}px`;
+      }
+    }
+
+    return null;
+  };
```

Comment:
You should no longer need this logic with the ResizeObserver being in the DataGrid component.  You should be able to accept the table height as a prop and use directly.  It will also allow you to remove the polyfill.

Reply:
Oh i didn't commit those changes. Committed here: 82615a08708ec4735399361487866e70b9601cff